Develop3

amechan92 opened the pull request:
Добавлены UI Тесты, добавлен Allure-отчёт

amechan92 committed:
Добавлены UI Тесты

amechan92 committed:
Добавлены UI Тесты

amechan92 committed:
Добавлен Allure-отчет

mkouznetsov-git commented on src/test/java/tests/LoginTests.java:
```diff
@@ -0,0 +1,80 @@
+import pageobjects.PasswordRecoveryPage;
+import pageobjects.ProfilePage;
+import pageobjects.RegistrationPage;
+import utils.WebDriverFactory;
```
Нет такого пакета и такого класса в проекте

amechan92 replied:
Добавила потерянный пакет utils в коммите 2f9791073faba30167469f4900b8eef6c23e2d4e

amechan92 committed:
Добавлен пакет utils, внесены изменения в pom.xml для Allure Maven Plugin

amechan92 committed:
Delete src/main/java/org/example directory

Удалена директория org/example

mkouznetsov-git commented on src/test/java/tests/LoginTests.java:
```diff
@@ -0,0 +1,80 @@
+        loginPage.login("himitsu1892@gmail.com", "123456");
+        mainPage.goToProfile();
+        WebDriverWait wait = new WebDriverWait(driver, 10);
+        wait.until(ExpectedConditions.elementToBeClickable(profilePage.getLogoutButton())).click();
```
В этом месте тест падает:

org.openqa.selenium.NoSuchElementException: no such element: Unable to locate element: {"method":"xpath","selector":"//*[@id="root"]/div/main/div/nav/ul/li[3]/button"}

amechan92 replied:
Добавила задержку в 2 секунды.

mkouznetsov-git commented on src/test/java/tests/MainPageTests.java:
```diff
@@ -0,0 +1,55 @@
+
+    @Test
+    public void testBunTabOpens() {
+       /* mainPage.clickSauceTab();
```
По заданию надо проверить именно переход на вкладку.

amechan92 replied:
Раскомментировала код

mkouznetsov-git commented on src/test/java/tests/RegistrationTests.java:
```diff
@@ -0,0 +1,50 @@
+        String emailDomain = "example.com";
+        String email = emailPrefix + UUID.randomUUID().toString().substring(0, 8) + "@" + emailDomain;
+        registrationPage.register(username, email, "password123");
+        assertTrue("The user should be registered successfully", registrationPage.isSuccessfulRegistration());
```
То, что пользователя перекидывает на страницу входа -- это ещё не значит, что регистрация прошла успешно. Надо конкретизировать тест.

amechan92 replied:
Добавила авторизацию в тесте после успешной регистрации для подтверждения.

mkouznetsov-git commented on src/main/java/pageobjects/RegistrationPage.java:
```diff
@@ -0,0 +1,81 @@
+
+public class RegistrationPage {
+    private WebDriver driver;
+    private By passwordError = By.xpath("//*[contains(@class,'input__error')]");
```
Мы проверяем интерфейс пользователя, поэтому мы должны по максимуму ориентироваться на то, что видит пользователь. Значит, нужно ориентироваться на текст, а не на класс элемента.

amechan92 replied:
Изменила на проверку по тексту "Некорректный пароль"

amechan92 committed:
Исправления:
Добавила задержку в 2 секунды для LoginTests
Раскомментировала код в MainPageTests
Добавила авторизацию в тесте после успешной регистрации для подтверждения в RegistrationTests
Изменила икспас на проверку по тексту "Некорректный пароль" в RegistrationPage

amechan92 committed:
Merge remote-tracking branch 'origin/develop3' into develop3
